include/beast/websocket/impl/stream.ipp

```diff
@@ -170,18 +170,29 @@ template<class NextLayer>
 template<class... Args>
 stream<NextLayer>::
 stream(Args&&... args)
-    : next_layer_(std::forward<Args>(args)...)
-    , stream_(next_layer_)
+    : stream_(std::forward<Args>(args)...)
 {
-    static_assert(is_Stream<next_layer_type>::value,
-        "Stream requirements not met");
+}
+
+template<class NextLayer>
+void
+stream<NextLayer>::
+accept()
+{
+    static_assert(is_SyncStream<next_layer_type>::value,
+        "SyncStream requirements not met");
```

Comment:
Are the `static_assert`s redundant in the overloads that throw, since they call overloads that don't throw, which have duplicate checks? I suppose it doesn't hurt since they are static after all, but I figured it's worth asking.